lib/workers/repository/process/lookup/index.ts

```diff
@@ -82,6 +83,12 @@ export async function lookupUpdates(
     res.homepage = dependency.homepage;
     res.changelogUrl = dependency.changelogUrl;
     res.dependencyUrl = dependency?.dependencyUrl;
+
+    if (!updatePinnedDependencies && versioning.isSingleVersion(currentValue)) {
```

Comment:
> I think if we have this new option `updatePinnedDependencies` (defaulting to `true`) then we can add a step prior to lookup which says that if `versioning.isSingleVersion(currentValue)` then disable updates.

@rarkins can you check if the code does this behavior?

Reply:
I must say it's clearly not exactly prior to it, but I put it after the lookup itself so that possible lookup errors can be traced and metadata can be populated(though I'm not sure it will be used in this case). If it's not important to keep this behaviour, it can be put prior to lookup to save some requests.

Reply:
For now I think it's best if we put the logic prior to lookup, and set a new "skip reason" value.

Longer term my goal - not just for this option/feature but for many others - is that we go through with the lookup and sorting and then mark the *updates* as skipped rather than the *dependency*. This would give greater visibility, e.g. you could see "you have an update pending but you've disabled it" instead of just "you have disabled this dependency".